src/RCP.h

```diff
@@ -0,0 +1,12 @@
+#ifndef BONDGRAPH_RCP_H_
+#define BONDGRAPH_RCP_H_
+
+#ifdef THIRDPARTY_TEUCHOS
+#    include "thirdparty/teuchos/Teuchos_RCP.hpp"
+#    define RCPLIB Teuchos
+#else
+#    include <symengine/symengine_rcp.h>
+#    define RCPLIB SymEngine
+#endif
```

Comment:
Oh, so Teuchos is optional? Why bother with it then? Why make things unnecessarily "complicated"? Why not simply use SymSengine's RCP since you need SymEngine anyway?